### PR TITLE
TASK: Introduce full behat test for FrontendNodeController

### PR DESCRIPTION
**Pull request by mhsdesign**
Predecessor for #5304 
Fixes parts of https://github.com/neos/neos-development-collection/issues/4933

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Tests/Behavior/Features/Bootstrap/DispatcherTrait.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    public function setupDispatcherTest(): void
+    {
+        $this->getObject(ContentCache::class)->flush();
+        $this->response = null;
+    }
+
+    /**
+     * @When the sites Fusion code is:
+     */
+    public function iHaveTheFollowingFusionCodeForTheSite(PyStringNode $fusionCode)
```
Comment (mhsdesign) [inline Neos.Neos/Tests/Behavior/Features/Bootstrap/DispatcherTrait.php]: 😭 😭 the new `unsafeSetAdditionalFusionSourceCodeToThisSingleton` is exactly what we NOT want imo...

File: Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php [hunk 2]
```diff
@@ -52,12 +52,6 @@ public static function tryFromFilePath(string $filePath): self
         return self::fromFilePath($filePath);
     }
 
-    public static function tryFromPackageRootFusion(string $packageKey): self
-    {
-        $fusionPathAndFilename = sprintf('resource://%s/Private/Fusion/Root.fusion', $packageKey);
-        return self::tryFromFilePath($fusionPathAndFilename);
-    }
```
Comment (mhsdesign) [inline Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php]: this method was removed as its considered internal and relatively new code from myself :D
The `FusionAutoIncludeHandler` will handle that now.

File: Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php [hunk 3]
```diff
@@ -50,19 +53,12 @@ public function createFromAutoIncludes(): FusionSourceCodeCollection
...
             }
         }
         return $sourcecode;
     }
 
-    public function createFromSite(Site $site): FusionSourceCodeCollection
-    {
-        return FusionSourceCodeCollection::tryFromPackageRootFusion($site->getSiteResourcesPackageKey());
-    }
-
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php]: this method was removed as its considered internal and relatively new code from myself :D

It was a yagni thing and we didn't even use it once :D

Comment (mhsdesign) [reply Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php]: @kitsunet if you would prefer rather a deprecation here i would also be fine with that ...

File: Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php (hunk 2 above)
Comment (kitsunet) [reply Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php]: It is in 8.3 and 8.4 though.... so uhhh no?

File: Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php (hunk 3 above)
Comment (kitsunet) [reply Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php]: def deprecation yeah, see also other comment above.

**Comment by kitsunet** [comment]
see above, deprecations instead of removals, rest is nice, although not sure why this is urgent/important atm?

File: Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php (hunk 3 above)
Comment (mhsdesign) [reply Neos.Neos/Classes/Domain/Service/FusionSourceCodeFactory.php]: 👍 did it :)

**Comment by mhsdesign** [comment]
Otherwise the pile im building upon this gets too large and i wanted to have that merged since the sprint. Thanks :) (see https://github.com/neos/neos-development-collection/pull/5304)

File: Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php [hunk 4]
```diff
@@ -52,6 +52,9 @@ public static function tryFromFilePath(string $filePath): self
         return self::fromFilePath($filePath);
     }
 
+    /**
+     * @deprecated with Neos 9, remove me :)
```
Comment (bwaidelich) [inline Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php:56]: why, what will it be replaced with?

**Review by bwaidelich — APPROVED** [review]
Great, thanks for taking care.
just a mini comment re `@deprecation` comment without details – feel free to ignore

File: Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php (hunk 4 above)
Comment (mhsdesign) [reply Neos.Fusion/Classes/Core/FusionSourceCodeCollection.php:56]: this is not api anymore with Neos 9.0 and you can make it yourself ... there is just no need for this as we dont use it for the normal fusion view neither. The source code collection should not deal with this logic and be flow package aware.

**Pull request merged**